### DISCUSSION TITLE
how can I detect newline ("\n")?

### DISCUSSION THREAD

**Discussion by jarodtang** [petitparser/dart-petitparser#136] [Q&A] [answered]
Hi there,

I craft some grammar like
'''
  Parser term() => ref0(character).plus().flatten().trim();

  ///
  Parser character() =>
      ref0(characterNormal) | ref0(characterUnicode) | ref0(characterEscape);
  Parser characterNormal() => pattern('^ ,.{}[]|=#*/_\r\n\\');
  Parser characterEscape() => char('\\'); 
  Parser characterUnicode() => string('\\u') & pattern('0-9A-Fa-f').times(4);
''

and try parse something like "a\nb", where I try to detect \n as new line, but the parser always ignore the '\n', how can I handle it?

Regards,
Jarod

**Comment by renggli** [comment, marked as answer]
Note that `.trim()` consumes all whitespace (including new-lines), so that might be a problem?

**Reply by jarodtang** [reply]
> Note that `.trim()` consumes all whitespace (including new-lines), so that might be a problem?

It's the problem! Thanks!
